controllers/secret_controller.go

```diff
@@ -258,6 +260,11 @@ func (s *SecretSyncController) Reconcile(ctx context.Context, req ctrl.Request)
 }
 
 func requiresUpdate(source, dest client.Object) bool {
+	// In case we target the same cluster
```

Comment:
We discussed this, and sounds to be unique based on generated UID for source and destination.